File: src/test/java/com/leetcode/P300Test.java
package com.leetcode;

import org.junit.jupiter.api.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

class P300Test {
    private final P300 sut = new P300();

    @Test
    void example01() {
        assertThat(sut.lengthOfLIS(new int[]{10, 9, 2, 5, 3, 7, 101, 18}), is(4));
    }

    @Test
    void example02() {
        assertThat(sut.lengthOfLIS(new int[]{0, 1, 0, 3, 2, 3}), is(4));
    }

    @Test
    void example03() {
        assertThat(sut.lengthOfLIS(new int[]{7, 7, 7, 7, 7, 7, 7}), is(1));
    }

    @Test
    void example04() {
        assertThat(sut.lengthOfLIS(new int[]{1, 3, 6, 7, 9, 4, 10, 5, 6}), is(6));
    }

}

File: src/test/kotlin/com/leetcode/P300Test.kt
package com.leetcode

import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.jupiter.api.Test

internal class P300Test {
    private val sut = P300()

    @Test
    fun `example 1`() {
        MatcherAssert.assertThat(sut.lengthOfLIS(intArrayOf(10, 9, 2, 5, 3, 7, 101, 18)), Matchers.`is`(4))
    }

    @Test
    fun `example 2`() {
        MatcherAssert.assertThat(sut.lengthOfLIS(intArrayOf(0, 1, 0, 3, 2, 3)), Matchers.`is`(4))
    }

    @Test
    fun `example 3`() {
        MatcherAssert.assertThat(sut.lengthOfLIS(intArrayOf(7, 7, 7, 7, 7, 7, 7)), Matchers.`is`(1))
    }

    @Test
    fun `example 4`() {
        MatcherAssert.assertThat(sut.lengthOfLIS(intArrayOf(1, 3, 6, 7, 9, 4, 10, 5, 6)), Matchers.`is`(6))
    }
}